scipy/stats/_distn_infrastructure.py

```diff
@@ -2212,21 +2239,36 @@ def fit(self, data, *args, **kwds):
               function to be optimized) and ``disp=0`` to suppress
               output as keyword arguments.
 
+            - method : The method to use. The default is "MLE" (Maximum
+              Likelihood Estimate); "GMM" (Generalized Method of Moments)
+              is also available.
+
+            - order : The number of moments to fit with ``method="GMM"``.
+              This integer must be greater than or equal to the number of
+              non-fixed distribution parameters (including location and scale);
+              the default is the number of non-fixed parameters. This keyword
+              is ignored when ``method="MLE"``.
+
         Returns
         -------
-        mle_tuple : tuple of floats
-            MLEs for any shape parameters (if applicable), followed by those
+        parameter_tuple : tuple of floats
+            Estimates for any shape parameters (if applicable), followed by those
             for location and scale. For most random variables, shape statistics
             will be returned, but there are exceptions (e.g. ``norm``).
 
         Notes
         -----
-        This fit is computed by maximizing a log-likelihood function, with
-        penalty applied for samples outside of range of the distribution. The
-        returned answer is not guaranteed to be the globally optimal MLE, it
+        With ``method="MLE"`` (default),
+        the fit is computed by maximizing a log-likelihood function, with
+        penalty applied for samples beyond the range of the distribution.
+
+        With ``method="GMM"``, the fit is computed by minimizing a weighted
+        norm of the errors between data moments and distribution moments.
```

Comment:
Will the weighted norm default to identity matrix if none is given? If so, may want to mention.

Reply:
I had planned on using [continuously updating GMM](https://en.wikipedia.org/wiki/Generalized_method_of_moments#Implementation) to estimate the appropriate weighting matrix with the parameters.

Reply:
ah, ok, I was assuming some constant matrix, seems that one has nice small sample properties so good call.

Reply:
It turns out our [proposal](https://warrenweckesser.github.io/czi/scipy-2019-czi-proposal-revised.pdf) only calls for regular, not generalized, method of moments, so I decided not to  generalize right now. Consequently, I changed the docstring:
```
        With ``method="MM"``, the fit is computed by minimizing the L2 norm
        of the errors between the first *k* data moments and the corresponding
        distribution moments, where *k* is the number of non-fixed parameters. 
        Typically, this error norm can be reduced to zero.
```
I think the scaffold is there, so hopefully it won't be too hard to add in the future if there's time.

Reply:
>It turns out our proposal only calls for regular, not generalized, method of moments, so I decided not to generalize right now.

+1